File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{humidity_v2_bricklet::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Humidity Bricklet 2.0.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let h = HumidityV2Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current humidity.
    let humidity = h.get_humidity().recv()?;
    println!("Humidity: {} %RH", humidity as f32 / 100.0);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/humidity_v2_bricklet"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Humidity Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	h, _ := humidity_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current humidity.
	humidity, _ := h.GetHumidity()
	fmt.Printf("Humidity: %f %RH\n", float64(humidity)/100.0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
